src/plots/cartesian/select.js

```diff
@@ -208,21 +208,23 @@ module.exports = function prepSelect(e, startX, startY, dragOptions, mode) {
 
     dragOptions.doneFn = function(dragged, numclicks) {
         corners.remove();
-        throttle.clear(throttleID);
-
-        if(!dragged && numclicks === 2) {
-            // clear selection on doubleclick
-            outlines.remove();
-            for(i = 0; i < searchTraces.length; i++) {
-                searchInfo = searchTraces[i];
-                searchInfo.selectPoints(searchInfo, false);
-            }
+        throttle.done(throttleID).then(function() {
```

Comment:
we need to wait for any throttling-delayed move event before we process the mouseup event, so we have the correct data.